tests/test_json.ml

```diff
@@ -0,0 +1,53 @@
+let read_file fname =
+  Fpath.(v ".." / ".." / ".." / "tests" / fname)
+  |> Fpath.to_string |> Yojson.Basic.from_file
+
+let dune_release_testing_json = read_file "dune-release-testing.json"
+
+let archive_upload_url () =
+  let check json expected =
+    match Dune_release.Json.archive_upload_url json with
+    | Ok actual -> Alcotest.(check string) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json
+    "https://github.com/NathanReb/dune-release-testing/releases/download/v0.0.0/dummy-v0.0.0.tbz"
+
+let release_id () =
+  let check json expected =
+    match Dune_release.Json.release_id json with
+    | Ok actual -> Alcotest.(check int) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json 12789323
+
+(*
+let upload_url () =
+  let check json expected =
+    match Dune_release.Json.upload_url json with
+    | Ok actual -> Alcotest.(check string) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json ""
+
+let html_url () =
+  let check json expected =
+    match Dune_release.Json.html_url json with
+    | Ok (`Url actual) -> Alcotest.(check string) __LOC__ actual expected
+    | Ok `Already_exists ->
+        Alcotest.(check string) __LOC__ "ALREADY_EXISTS" expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json ""
+*)
```

Comment:
I'm looking for the full specification before being able to complete these ones.